### PR TITLE
Add build parameters in detailed component

### PR DESCRIPTION
**Pull request by adellinekania**
Extend `/rest/api/2/components/{component}/versions/{version}` end-point with build related data

### CODE REVIEW

**Comment by pgorbachev** [comment]
@adellinekania , 
1) could please add tests for the changed functionality
2) add an example of the new section to the Description

**Review by pgorbachev — CHANGES_REQUESTED** [review]
Tests should be added

File: components-registry-service-server/src/test/kotlin/org/octopusden/octopus/components/registry/server/ComponentsRegistryServiceControllerTest.kt [hunk 1]
```diff
@@ -393,6 +395,53 @@ class ComponentsRegistryServiceControllerTest : BaseComponentsRegistryServiceTes
...
+                    ToolDTO(
+                        name = "PowerBuilderCompiler170",
+                        escrowEnvironmentVariable = "PBC_BIN",
+                        sourceLocation = "\$env.PBC/170",
+                        targetLocation = "tools/auto_compiler"
+                    )
+                ),
+            )
+        }
+        Assertions.assertEquals(expectedComponent, actualComponent)
```
Comment (pgorbachev) [inline components-registry-service-server/src/test/kotlin/org/octopusden/octopus/components/registry/server/ComponentsRegistryServiceControllerTest.kt:442]: Please check only buildParameters DTO. Let's make test isolated from other parameters

File: test-common/src/test/resources/components-registry/common/TestComponents.groovy [hunk 2]
```diff
@@ -840,3 +840,16 @@ TEST_COMPONENT4 {
         tag = 'tdsecure-$version'
     }
 }
+
+"TEST_COMPONENT_BUILD_PARAMETERS" {
```
Comment (pgorbachev) [inline test-common/src/test/resources/components-registry/common/TestComponents.groovy:844]: COMPONENT_WITH_BUILD_PARAMETERS

**Review by pgorbachev — APPROVED** [review]

**Review by ozonophore — APPROVED** [review]

File: components-registry-service-core/src/main/kotlin/org/octopusden/octopus/components/registry/core/dto/ToolDTO.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package org.octopusden.octopus.components.registry.core.dto
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class ToolDTO @JsonCreator constructor(
+    @JsonProperty("name") val name: String? = null,
```
Comment (aryabokon) [inline components-registry-service-core/src/main/kotlin/org/octopusden/octopus/components/registry/core/dto/ToolDTO.kt:9]: can we make this properties as non-nulable?

Comment (adellinekania) [reply components-registry-service-core/src/main/kotlin/org/octopusden/octopus/components/registry/core/dto/ToolDTO.kt:9]: Yes, it is possible to make the name property non-nullable, as the corresponding `Tool` data class in the `component-resolver-api` already defines it as non-nullable. However, I would need confirmation on whether every Tool instance is guaranteed to have a name. If so, I can safely enforce non-nullability here.

Comment (aryabokon) [reply components-registry-service-core/src/main/kotlin/org/octopusden/octopus/components/registry/core/dto/ToolDTO.kt:9]: @adellinekania , you need to have guarantee via validation on EscrowConfig loading

@pgorbachev , what about another fields? Do we have a contract?

Comment (pgorbachev) [reply components-registry-service-core/src/main/kotlin/org/octopusden/octopus/components/registry/core/dto/ToolDTO.kt:9]: @adellinekania , 
All the parameters are mandatory except install script.
So 
1) Add validateBuildParameters to EscrowConfigValidator.groovy to ensure this (add test for this in EscrowConfigurationLoaderTest)
2) make DTO fields non-null except install script

File: component-resolver-core/src/main/groovy/org.octopusden/octopus/escrow/configuration/validation/EscrowConfigValidator.groovy [hunk 4]
```diff
@@ -411,6 +412,25 @@ class EscrowConfigValidator {
         }
     }
 
+    def validateBuildConfigurationTools(EscrowModuleConfig moduleConfig) {
+        def tools = moduleConfig.getBuildConfiguration()?.getTools()
+        tools?.each { tool ->
+            def toolName = tool.getName()
+            if (toolName== null)  {
```
Comment (pgorbachev) [inline component-resolver-core/src/main/groovy/org.octopusden/octopus/escrow/configuration/validation/EscrowConfigValidator.groovy:419]: Please, reformat the change code

**Review by pgorbachev — DISMISSED** [review]

File: component-resolver-core/src/main/groovy/org.octopusden/octopus/escrow/configuration/validation/EscrowConfigValidator.groovy [hunk 5]
```diff
@@ -411,6 +412,25 @@ class EscrowConfigValidator {
         }
     }
 
+    def validateBuildConfigurationTools(EscrowModuleConfig moduleConfig) {
+        def tools = moduleConfig.getBuildConfiguration()?.getTools()
+        tools?.each { tool ->
+            def toolName = tool.getName()
+            if (toolName == null)  {
+                registerError("tool is not configured correctly")
```
Comment (CamaradeRoman) [inline component-resolver-core/src/main/groovy/org.octopusden/octopus/escrow/configuration/validation/EscrowConfigValidator.groovy:420]: let's change error description to more specific, like "tool name is not specified"

**Review by pgorbachev — APPROVED** [review]

**Review by CamaradeRoman — APPROVED** [review]

File: components-registry-service-core/src/main/kotlin/org/octopusden/octopus/components/registry/core/dto/ToolDTO.kt (hunk 3 above)
Comment (aryabokon) [reply components-registry-service-core/src/main/kotlin/org/octopusden/octopus/components/registry/core/dto/ToolDTO.kt]: @adellinekania , we talked about nonNullable fields and fields still nullable

**Review by aryabokon — CHANGES_REQUESTED** [review]
deleted

**Pull request merged**